src/tracker-component.js

```diff
@@ -343,7 +389,18 @@ export const Tracker = (config? : Config = defaultTrackerConfig) => {
             };
             track(config, 'setUser', { oldUserId: getUserIdCookie() });
         },
-        setPropertyId: (id : string) => {
+        setPropertyId: (id : string, automatic : boolean = false) => {
```

Comment:
I'm not a fan of the pattern of having a single method that changes behavior based on a flag that's passed in, I think it increases complexity both on development and usage of the method. Would it make sense to just create another method? We can keep `setPropertyId` as it was originally, and then have a separate internal method for the new stuff.